### PR TITLE
expose s3 bucket lifecycle_rule_enabled

### PR DESCRIPTION
**Pull request by deadlysyn**
Would like to bubble up terraform-aws-s3-log-storage's lifecycle setting so it can easily be controlled in terraform-aws-alb (per https://github.com/cloudposse/terraform-aws-alb/issues/39).

### CODE REVIEW

**Comment by jamengual** [comment]
@deadlysyn do you want to expose the boolean or you are interesting on exposing more options ? https://github.com/cloudposse/terraform-aws-s3-log-storage/blob/master/main.tf#L25

**Comment by deadlysyn** [comment]
i am open to taking this PR in the direction most helpful to the community...  do you think we should expose more?

i minimized scope of change initially because the defaults from terraform-aws-s3-log-storage worked for me. if you think it's better, we can bubble everything up w/ same defaults so it's all tunable but nothing new is required for existing uses.

**Comment by deadlysyn** [comment]
@jamengual see if this looks better -- to keep the interface as simple as possible, i've maintained some defaults but bubbled up tier/retention-related settings.

File: main.tf [hunk 1]
```diff
@@ -38,20 +38,26 @@ data "aws_iam_policy_document" "default" {
...
+  policy                             = join("", data.aws_iam_policy_document.default.*.json)
+  force_destroy                      = var.force_destroy
+  versioning_enabled                 = true
+  lifecycle_rule_enabled             = var.lifecycle_rule_enabled
+  enable_glacier_transition          = var.enable_glacier_transition
+  expiration_days                    = var.expiration_days
+  glacier_transition_days            = var.glacier_transition_days
+  noncurrent_version_expiration_days = var.noncurrent_version_expiration_days
+  noncurrent_version_transition_days = var.noncurrent_version_transition_days
+  standard_transition_days           = var.standard_transition_days
```
Comment (jamengual) [inline main.tf:58]: these options are not exposed in the terraform-aws-s3-log-storage so test is failing, so there is more work to do upstream, I will work on PR today

**Comment by osterman** [comment]
Tests are failing...
image

File: main.tf (hunk 1 above)
Comment (jamengual) [reply main.tf:58]: I was wrong, they are exposed but the module version source needs to be updated @deadlysyn

**Comment by jamengual** [comment]
@deadlysyn can you update to the latest version 0.11.0 please , thanks

File: main.tf [hunk 2]
```diff
@@ -38,20 +38,26 @@ data "aws_iam_policy_document" "default" {
...
-  acl                    = var.acl
-  policy                 = join("", data.aws_iam_policy_document.default.*.json)
-  force_destroy          = var.force_destroy
-  versioning_enabled     = true
-  lifecycle_rule_enabled = false
-  delimiter              = var.delimiter
-  attributes             = var.attributes
-  tags                   = var.tags
-  lifecycle_prefix       = var.lifecycle_prefix
+  source                             = "git::https://github.com/cloudposse/terraform-aws-s3-log-storage.git?ref=tags/0.9.0"
```
Comment (jamengual) [inline main.tf:41]:
```suggestion
  source                             = "git::https://github.com/cloudposse/terraform-aws-s3-log-storage.git?ref=tags/0.11.0"
```

**Comment by deadlysyn** [comment]
hi folks, i don't have full context on the test env (but it looks pretty cool!)... since it seems to fail on creating the bucket due to resource/naming conflict (bucket already exists?) i wonder if we could create a random string w/ terraform, use it as part of the bucket name (attributes?), and then retrieve from state so terratest can validate (vs hard-coded name).

that might not work, it's just an idea on how to avoid the collision...if it is failing to cleanup for reasons we can control, happy to accept input on how existing solution can be improved (don't want to leave junk behind). if it is conflict with other users/tests, then the randomized name might help. i can work on that if it's a good solution.

**Comment by jamengual** [comment]
> hi folks, i don't have full context on the test env (but it looks pretty cool!)... since it seems to fail on creating the bucket due to resource/naming conflict (bucket already exists?) i wonder if we could create a random string w/ terraform, use it as part of the bucket name (attributes?), and then retrieve from state so terratest can validate (vs hard-coded name).
> 
> that might not work, it's just an idea on how to avoid the collision...if it is failing to cleanup for reasons we can control, happy to accept input on how existing solution can be improved (don't want to leave junk behind). if it is conflict with other users/tests, then the randomized name might help. i can work on that if it's a good solution.

we are aware of the issue and hopefully, we should have a solution soon

**Comment by osterman** [comment]
@deadlysyn please rebase off `master` so we can pull in the fixes to the tests

**Pull request merged**